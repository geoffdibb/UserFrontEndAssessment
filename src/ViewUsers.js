import React, { Component } from 'react';
import Users from "./Users";
import axios from 'axios';
import { Table } from 'reactstrap';


export default class viewUsers extends Component {

    constructor() {
        super();
        this.state = {
            data: []
        }
    }

    getall = () => {

        axios
            .get("http://localhost:5000/User/all")
            .then(response => {
                this.setState({
                    data: response.data
                });
            }).catch(err => console.log(err));
    };

    componentDidMount() {
        this.getall();
    }


    render(props) {
        return (
            <div class="formDiv">
                <div id="random"> </div>
                <h2>Users</h2>
                <Table>
                        {this.state.data.map((item) => (

                                 <Users key={item.username} passedFunction={this.props.passedFunction} username={item.username}/>       

                        ))}
                </Table>
            </div>

        );
    }
}

