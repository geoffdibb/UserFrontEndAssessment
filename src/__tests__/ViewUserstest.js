import React from 'react';
import ReactDOM from 'react-dom';
import ViewUsers from '../ViewUsers';
import renderer from 'react-test-renderer';


it('snapshot ViewUsers', () => {
  const tree = renderer.create(<ViewUsers/>).toJSON()
  expect(tree).toMatchSnapshot();
})