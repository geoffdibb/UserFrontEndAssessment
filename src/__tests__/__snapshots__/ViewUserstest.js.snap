// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot ViewUsers 1`] = `
<div
  class="formDiv"
>
  <div
    id="random"
  >
     
  </div>
  <h2>
    Users
  </h2>
  <table
    className="table"
  />
</div>
`;
